pkg/autogen/autogen.go

```diff
@@ -240,3 +242,96 @@ func GenerateRulePatches(spec *kyverno.Spec, controllers string, log logr.Logger
 	}
 	return
 }
+
+type Policy interface {
+	GetAnnotations() map[string]string
+	GetSpec() kyverno.Spec
+}
+
+// GenerateRules generates rule for podControllers based on scenario A and C
+func GenerateRules(spec *kyverno.Spec, controllers string, log logr.Logger) []kyverno.Rule {
+	var rules []kyverno.Rule
+	for _, rule := range spec.Rules {
+		// handle all other controllers other than CronJob
+		genRule := generateRuleForControllers(*rule.DeepCopy(), stripCronJob(controllers), log)
+		if genRule != nil {
+			rules = append(rules, convertRule(*genRule, "Pod"))
+		}
+		// handle CronJob, it appends an additional rule
+		genRule = generateCronJobRule(*rule.DeepCopy(), controllers, log)
+		if genRule != nil {
+			rules = append(rules, convertRule(*genRule, "Cronjob"))
+		}
+	}
+	return rules
+}
+
+func convertRule(rule kyvernoRule, kind string) kyverno.Rule {
+	// TODO: marshall, rewrite and unmarshall
+	if bytes, err := json.Marshal(rule); err != nil {
+		// TODO
+	} else {
+		bytes = updateGenRuleByte(bytes, kind, rule)
+		if err := json.Unmarshal(bytes, &rule); err != nil {
+			// TODO
+		}
+	}
```

Comment:
How do we handle errors?

Reply:
Ah yes, I think I left a couple `TODO`s behind.

Reply:
Fixed.